src/web/metrics.rs

```diff
@@ -32,6 +39,52 @@ fn duration_to_seconds(d: Duration) -> f64 {
     d.as_secs() as f64 + nanos
 }
 
+/// Request recorder middleware
+///
+/// Looks similar, but *is not* a usable middleware / layer
+/// since we need the route-name.
+///
+/// Can be used like:
+/// ```ignore
+/// get(handler).route_layer(middleware::from_fn(|request, next| async {
+///     request_recorder(request, next, Some("static resource")).await
+/// }))
+/// ```
+pub(crate) async fn request_recorder<B>(
+    request: AxumRequest<B>,
+    next: Next<B>,
+    route_name: Option<&str>,
+) -> impl IntoResponse {
+    let route_name = if let Some(rn) = route_name {
+        Cow::Borrowed(rn)
+    } else if let Some(path) = request.extensions().get::<MatchedPath>() {
+        Cow::Owned(path.as_str().to_string())
+    } else {
+        Cow::Owned(request.uri().path().to_string())
```

Comment:
what's the difference between these two?

Reply:
`MatchedPath` will be the route name including placeholders, `request.uri()` will be the full actual path. 

Example: for `get_internal` we don't have a fixed route name for tracking, there we want the dynamic route path as in the router.

Reply:
see also: https://docs.rs/axum/latest/axum/extract/struct.MatchedPath.html